src/core/channelz/channelz_registry.h

```diff
@@ -45,37 +45,37 @@ class ChannelzRegistry final {
   static void Unregister(BaseNode* node) {
     Default()->InternalUnregister(node);
   }
-  static RefCountedPtr<BaseNode> Get(intptr_t uuid) {
+  static WeakRefCountedPtr<BaseNode> Get(intptr_t uuid) {
```

Comment:
Hmm... It feels a little odd to expose weak refs here, since those are really intended to be internal.

Are there going to be any cases where there's more than one strong ref held to a given object?  If not, I wonder if it would make sense to use `InternallyRefCounted<>` instead of `DualRefCounted<>`?

Reply:
It's common to share a node across different layers implementing that object (and indeed is the magic that allows DataSource to exist).